pytorch_lightning/plugins/__init__.py

```diff
@@ -47,3 +48,39 @@
     'DDPShardedPlugin',
     'DDPSpawnShardedPlugin',
 ]
+
+REGISTRY_TRAINING_TYPE_PLUGINS = [{
```

Comment:
Hey Kaushik! I assume you're just trying things out atm and this will eventually live in the corresponding plugins, but have a look at https://github.com/facebookresearch/ClassyVision/blob/040e51d993b96fa63a9a1e14a827604ab566d910/classy_vision/generic/registry_utils.py#L14 to go through a directory (`pytorch_lightning.plugins`) and automatically find the modules. This way we should be able to do something like `Plugin.register_plugins(registry)` and have each plugin do it for us, or automatically via a context manager when we import

Reply:
Got it, thanks! Doable. 
Yeah, just trying out stuff lol

Reply:
already making great headway!